### ISSUE TITLE
Member registration payment confirmation 500 error

### ISSUE THREAD

**Issue by Komkommer123** [svthalia/concrexit#1930]
### Describe the bug
When confirming a payment for a new member, randomly there will be 500 errors. Sometimes 3 after each other, most of the times no problem.

### How to reproduce
Steps to reproduce the behaviour:
1. Go to a member on the registration page
2. Confirm their payment, for example payment by card

### Expected behaviour
Confirmation of payment accepted

**Comment by se-bastiaan** [comment]
I think this may be connected to #945.

What seems to happen here is that there is no username in the registration when the payment is processed. That is weird because that value should be filled when the registration is accepted already: https://github.com/svthalia/concrexit/blob/bc99ae7573320bd2465d4686a1508a214b92f1f7/website/registrations/services.py#L153

Therefore that means that somehow the field is empty, but I have no idea how or why. We'd have to figure that out.

So TLDR:
The problem is that there is no username in the registration even tho there should be.

**Comment by DeD1rk** [comment]
There still is some kind of problem: https://thalia.sentry.io/issues/4409860701/.
